### ISSUE TITLE
CLI Support: Error when building with angular-cli (v.1.0.0-beta.24)

### ISSUE THREAD

**Issue by AnianZ** [ngrx/store#305]
I tried to use ngrx/store in a project build with the angular-cli (v1.0.0-beta.24) but it generates an error when building. To make sure the problem is not in my project I created a blank project with `ng init` and just copied the example code from the ngrx README but it sill shows the same error. The weird thing is, that the app seems to work just fine. I can use the store in both cases. But the error is still annoying. Is ngrx/store supported by the cli? Can this error be fixed? As far as I understood it, it might be something concerning AoT compilation.

The Error message:
````
ERROR in Error encountered resolving symbol values statically. Function calls are not supported. Consider replacing the function or lambda with a reference to an exported function (position 8:54 in the original .ts file), resolving symbol counterReducer in F:/Dokumente/ngrx/test/src/app/counter.ts, resolving symbol AppModule in F:/Dokumente/ngrx/test/src/app/app.module.ts, resolving symbol AppModule in F:/Dokumente/ngrx/test/src/app/app.module.ts, resolving symbol AppModule in F:/Dokumente/ngrx/test/src/app/app.module.ts
````

**Comment by simb** [comment]
I would guess that to use the current version of ngrx/store you would need to be on at least RC5.  There are a ton of changes between rc4 and final having to do with Modules, Forms, Routing.  You should upgrade as soon as possible as none of the existing tools for Angular 2 will work with early betas.

**Comment by AnianZ** [comment]
Ok. Sorry, that was a bit ambiguous. I'm currently using angular at version 2.3.1 and ngrx/store at version 2.2.1. The angular-cli is at version 1.0.0-beta.24, the newest version to date.

**Comment by littleStudent** [comment]
take a look at this issue. https://github.com/ngrx/store/issues/190
you should find the solution there.

**Comment by AnianZ** [comment]
Thanks. It's seems it is as simple as using an untyped `function` instead of a typed `const` with a lambda. At least in the example case.
So this is for compatibility with AoT and the CLI only works with AoT compatible code? Maybe we should adapt the examples in the ngrx documentation to avoid confusion for beginners. For me this was a quite confusing error.

**Comment by AnianZ** [comment]
I added a small fix to the README #306
